src/Slider.jsx

```diff
@@ -339,7 +339,7 @@ class Slider extends React.Component {
     const lowerOffset = this.calcOffset(lowerBound);
 
     const handleClassName = prefixCls + '-handle';
-    const isNoTip = (step === null) && !tipFormatter;
+    const isNoTip = tipFormatter === null || (step === null && !tipFormatter);
```

Comment:
Was being as careful as possible with introducing a breaking change. But I'm fine with this if you are.